master/networking/ip-autodetection.md

```diff
@@ -0,0 +1,199 @@
+    ipv4Address: 10.244.0.1/24
+    ipv6Address: 2000:db8:85a3::8a2e:370:7335/120
+    ipv4IPIPTunnelAddr: 192.168.0.1
+```
+
+#### Auto detection methods
+
+By default, {{site.prodname}} uses the **first-found** method; the first valid IP address on the first interface (excluding local interfaces such as the docker bridge). However, you can change the default method to any of the following:
+
+- Address used by the node to reach a particular IP or domain (**can-reach**)
+- Regex to include matching interfaces (**interface**)
+- Regex to exclude matching interfaces (**skip-interface**)
+
+For details on auto detection methods, see [node configuration]({{site.baseurl}}/{{page.version}}/reference/node/configuration#ip-autodetection-methods) reference.
+
+#### Manually configure IP address and subnet
+
+There are two ways to manually configure an IP address and subnet:
+
+- {{site.prodname}} node container (start/restart)
+  Use environment variables to set values for nodes.
+
+- {{site.prodname}} node resource
+  Update the node resource.
+
+##### Using environment variables and node resource
+
+Because you can configure IP address and subnet using either environment variables or node resource, the following table describes how values are synchronized.
+
+| **If this environment variable...** | **Is...**                                             | **Then...**                                                  |
+| ----------------------------------- | ----------------------------------------------------- | ------------------------------------------------------------ |
+| IP/IP6                              | Explicitly set                                        | The specified values are used, and the Node resource is updated. |
+|                                     | Set to autodetect                                     | The requested method is used (first-found, can-reach, interface, skip-interface), and the Node resource is updated. |
+|                                     | Not set, but Node resource has IP/IP6 values          | Node resource value is used.                                 |
+| IP                                  | Not set, and there is no IP value in Node resource    | Auto detects an IPv4 address and subnet, and updates Node resource. |
+| IP6                                 | Not set, and there is a notIP6 value in Node resource | No IP6 routing is performed on the node.                     |
+
+### How to
+
+- [Change the auto detection method](#change-the-auto-detection-method)
+- [Manually configure IP address and subnet for a node](#manually-configure-ip-address-and-subnet-for-a-node)
+
+#### Change the auto detection method
+
+As noted previously, the default auto detection method is **first valid interface found** (first-found). To use a different auto detection method, use the following `kubectl patch` command, specifying the method:
+
+- **IPv4**
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
```

Comment:
Why not use the `kubectl set env` command, which makes this a bit nicer looking and easier to follow? https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-env-em-

Reply:
Good point, fixed